service/controller/resource/nodepool/create_deployment_uninitialized.go

```diff
@@ -187,14 +194,9 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 	}
 }
 
-func (r *Resource) saveAzureIDsInCR(ctx context.Context, virtualMachineScaleSetsClient *compute.VirtualMachineScaleSetsClient, virtualMachineScaleSetVMsClient *compute.VirtualMachineScaleSetVMsClient, azureMachinePool *capzexpv1alpha3.AzureMachinePool) error {
+func (r *Resource) saveAzureIDsInCR(ctx context.Context, virtualMachineScaleSetsClient *compute.VirtualMachineScaleSetsClient, virtualMachineScaleSetVMsClient *compute.VirtualMachineScaleSetVMsClient, azureMachinePool *capzexpv1alpha3.AzureMachinePool, vmss compute.VirtualMachineScaleSet) error {
```

Comment:
vmss can be nil, does this even compile?

Reply:
I mean, shouldn't it be a pointer?

Reply:
Oh, now I got how it works.
Maybe I would use a pointer and set it to nil in case of 404, then check for pointer != nil where you want to use the `vmss`.
But I like the idea in general

Reply:
If we can use non-pointer value (especially given that e.g. autorest `Response` provides functionality), I feel it's better. Less unnecessary nil checks and less potential for null pointer dereferencing, but your call. I'm not blocking on this.

Reply:
Yeah, I'd rather not to deal with pointers.